File: DevExpress-Demo/VB/GridDemo.Wpf/ModuleResources/PrintTemplatesViewModel.vb
Namespace GridDemo

    Public Class PrintTemplatesViewModel
        Inherits PrintViewModelBase

        Public Property Mode As PrintTemplatesMode
            Get
                Return GetProperty(Function() Me.Mode)
            End Get

            Set(ByVal value As PrintTemplatesMode)
                SetProperty(Function() Mode, value)
            End Set
        End Property

        Public Overrides Sub ReOpenPreviewTabs()
            Mode = PrintTemplatesMode.Default
            MyBase.ReOpenPreviewTabs()
            Mode = PrintTemplatesMode.MailMerge
            ShowPreviewInNewTab()
            Mode = PrintTemplatesMode.Detail
            ShowPreviewInNewTab()
        End Sub

        Protected Overrides Function GetTitle() As String
            Return Mode.ToString() & " Print Preview"
        End Function
    End Class

    Public Enum PrintTemplatesMode
        Detail
        MailMerge
        [Default]
    End Enum
End Namespace

File: DevExpress-Demo/CS/GridDemo.Wpf/ModuleResources/PrintTemplatesViewModel.cs
using System;
using DevExpress.Mvvm.DataAnnotations;
using DevExpress.Xpf.Printing;

namespace GridDemo {
    public class PrintTemplatesViewModel : PrintViewModelBase {
        public PrintTemplatesMode Mode {
            get { return GetProperty(() => Mode); }
            set { SetProperty(() => Mode, value); }
        }

        public override void ReOpenPreviewTabs() {
            Mode = PrintTemplatesMode.Default;
            base.ReOpenPreviewTabs();
            Mode = PrintTemplatesMode.MailMerge;
            base.ShowPreviewInNewTab();
            Mode = PrintTemplatesMode.Detail;
            base.ShowPreviewInNewTab();
        }

        protected override string GetTitle() {
            return Mode.ToString() + " Print Preview";
        }
    }
    public enum PrintTemplatesMode {
        Detail,
        MailMerge,
        Default
    }
}
